src/ZPublisher/request_params.rst

```diff
@@ -0,0 +1,296 @@
+``:``\ *directive* to specify such directives. For example,
+the parameter ``i:int=1`` tells Zope to convert the value ``'1'`` to an
+integer and use it as value for request variable ``i``; the parameter sequence
+``x.name:record=Peter&x.age:int:record=10`` tells Zope to contruct
+a record ``x`` with attributes ``name`` and ``age`` and respective values ``'Peter'`` and ``10``.
+
+The remaining parts of this document describe the available kinds
+of directives and the overall processing model.
+
+
+Request parameter directive kinds
+=================================
+
+Zope supports different kinds of request parameter directives.
+This sections describes them in detail.
+
+
+Converter
+---------
+
+A converter directive converts the parameter value (a string) to a converter
+specific type in a coverter specific way. The available converters are given
+by ``ZPublisher.Converters.type_converters``.
+
+Simple type converters are ``int``, ``float``, ``boolean``, ...
+each converting to the respective type.
+
+Some converters employ special logic for the conversion.
+For example, both ``tokens`` as well as ``lines`` convert to
+a list of strings but ``tokens`` splits the input at whitespace, ``lines``
+at newlines.
+
+The converter ``required`` contains only logic: it checks that
+the value is not empty.
+
+Different Zope versions can support different converters.
+
+
+Encoding
+--------
+
+An encoding directive tells the converting process the encoding
+of the parameter value. Typical encodings are "utf8", "latin1" ...
+
+An encoding directive is ignored, if the parameter does not
+have a converter directive as well. In some sense, the encoding
+directive informs the converter about the value's encoding.
+If there is no encoding directive, the converter uses the
+default encoding (specified via the configuration option
+``zpublisher-default-encoding``).
+
+In principle, Zope supports any encoding known by the ``codecs``
+module. However, the converter may impose restrictions.
+
+
+Aggregator
+----------
+
+An aggregator directive tells Zope how to process parameters with
+a similar name.
+
+Zope supports the following aggregators:
+
+list
+  collect all values with this name into a list.
+
+  If there are more than a single parameter with the same name,
+  then they are by default collected into a list.
+  The ``list`` aggregator is mainly used to ensure that
+  the parameter leads to a list value even in the case that
+  there is only one of them.
+
+tuple
+  collect all values with this name into a tuple.
+
+default
+  use the value of this parameter as a default value; it
+  can be overridden by a parameter of the same name without
+  the ``default`` directive.
+
+record
+  this directive assumes that the parameter name starts with
+  *var*\ ``.``\ *attr*.
+
+  It tells Zope to create a request variable *var*
+  with as value a 
+  record (i.e. a ``ZPublisher.HTTPRequest.record`` instance) and
+  set its attribute *attr* to the parameter value.
```

Comment:
Sounds a bit rough.... maybe... "It tells Zope to create a request variable *var* of type record...